akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/JournalInteractions.scala

```diff
@@ -119,4 +119,50 @@ private[akka] trait JournalInteractions[C, E, S] {
         state.state), setup.selfUntyped)
   }
 
+  /**
+   * On [[akka.persistence.SaveSnapshotSuccess]], if [[EventSourcedSettings.retainSnapshotsOnly]]
+   * is enabled, this is not executed and continues to `internalDeleteSnapshots`, otherwise old
+   * messages are deleted based on [[EventSourcedSettings.snapshotSize]] before old snapshots are
+   * deleted based on [[EventSourcedSettings.keepNSnapshots]].
+   */
+  protected def internalDeleteEvents(e: SaveSnapshotSuccess, state: Running.RunningState[S]): Unit = {
```

Comment:
Unsure about this:

Reply:
General idea looks good to me. Or do you mean the details around how many events to keep etc?

Reply:
@johanandren I am unsure about the flow within the function specifically.